sql/core/src/main/scala/org/apache/spark/sql/execution/AggregatingAccumulator.scala

```diff
@@ -148,6 +148,14 @@ class AggregatingAccumulator private(
   }
 
   override def merge(other: AccumulatorV2[InternalRow, InternalRow]): Unit = withSQLConf(()) {
+    mergeInternal(other)
+  }
+
+  /**
+   * Actual code for merge. This function is not intended to be called directly from executor side.
+   * One exception is `addState`.
+   */
+  private def mergeInternal(other: AccumulatorV2[InternalRow, InternalRow]): Unit = {
```

Comment:
Why do we need to make a difference between merge and mergeInternal?

Reply:
According to the comment of `withSQLConf`, `merge` is usually intended to be called from driver side and not from executor side. In fact, `merge` with `withSQLConf` seems to separate logic for driver and executor. So, `mergeInternal` is for exceptional cases where executor really need to call the merge logic.

Reply:
> merge with withSQLConf seems to separate logic for driver and executor.

It's more like a sanity check to make sure `update/merge` is only called on the expected side (driver or executor). Since we now have a reason to call `merge` on the executor side, let's document it clearly and allow it. To keep `update` untouched, we can create an overload of `withSQLConf` for `merge`.